### ISSUE TITLE
problem of input size of mnist and svhn

### ISSUE THREAD

**Issue by Huang-yihao** [GuanlinLee/FPD-for-Adversarial-Robustness#1]
What's the input size of images for resnet101?
When I use size (64,64), I met the error below.
![image](https://user-images.githubusercontent.com/16575311/90972861-1e322000-e54f-11ea-91fd-f17b5402e092.png)
However, if I use size (220,220), then the accuracy of the model "ResNet-101-model-adv-pgd.pkl" is very low.
Therefore, could you please tell me what is the right setup for using the pre-trained model "ResNet-101-model-adv-pgd.pkl" provided by you?

Thank you very much!

**Comment by GuanlinLee** [comment]
The input size for MNIST and SVHN is (bs, 3, 64, 64). It seems like you do not use the proper model.py, as for ResNet-50 and ResNet-101 the restoration module has different channel sittings. More than that, please check your pytorch version, our experiments are under pytorch-1.1.0.

**Comment by Huang-yihao** [comment]
I am confused that if I add the below code into models.py.
![image](https://user-images.githubusercontent.com/16575311/91128697-35057d80-e6db-11ea-99d5-9392c8f21bbd.png)
The execution result is 
![image](https://user-images.githubusercontent.com/16575311/91128731-464e8a00-e6db-11ea-8dfe-df4e081ebf23.png)
While if I use size (64,64), the error is shown below.
![image](https://user-images.githubusercontent.com/16575311/91128867-8c0b5280-e6db-11ea-9754-7953ae9c7ee2.png)

Does it mean the input size of resnet101 model should be (220,220)? This is a conflict with the size(64,64) of MNIST and SVHN.

Could you please give an explanation?
Thank you !

**Comment by GuanlinLee** [comment]
As I commented before, you may use a wrong model.py. So the shape of final feature maps is too small to do an AVG-Pooling. Please check the shape of features and be sure you use the correct input shape and model file. All codes have been checked under pytorch-1.1.0.
If you have other questions, feel free to ask.

**Comment by Huang-yihao** [comment]
I run the code under pytorch-1.3.0. It may not influence a lot?

The main concern of mine can be directly summarized as below.
The test function in file  resnet.py of /FPD-for-Adversarial-Robustness/ResNet-101/MNIST/ use size(220,220) as input for resnet101. Why in file adv_train.py of /FPD-for-Adversarial-Robustness/ResNet-101/MNIST/ , it use size(64,64) as input for resnet101? The code of size(64,64) is shown below.
![image](https://user-images.githubusercontent.com/16575311/91143287-d8a75b80-e6e4-11ea-9ff0-8b0a4d3e3d97.png)

Could you please explain this ?
Thank you !

**Comment by GuanlinLee** [comment]
As we claimed in our paper, except for CALTECH-101 and CALTECH-256, all inputs are resized into (64,64). (220,220) is not a correct input size, so you obtain a low accuracy. The error is because the feature map has an incorrect shape as I have explained.

**Comment by Huang-yihao** [comment]
As in the test function, the size (220,220) is right for resent101. How to put image of size(64,64) into resnet101?

Thank you!

**Comment by GuanlinLee** [comment]
Sorry for the confusion. This function is just to validate the restoration shape same as the input. In /SVHN/resnet.py, the test function use the input shape (64,64). However, the resnet.py in /SVHN or /MNIST is total same. After all, it is just a test function. And our model should work when the input has shape (64,64). You may try to change the avgpool(7) to avgpool(2) if this works.

**Comment by Huang-yihao** [comment]
Thank you for your explanation, change to avgpool(2) is useful.
